diff --git a/cmake/ArrasNodeCompileOptions.cmake b/cmake/ArrasNodeCompileOptions.cmake
index 079f2ed..5316b47 100644
--- a/cmake/ArrasNodeCompileOptions.cmake
+++ b/cmake/ArrasNodeCompileOptions.cmake
@@ -12,7 +12,7 @@ function(ArrasNode_cxx_compile_options target)
                 -fno-omit-frame-pointer         # TODO: add a note
                 -fno-strict-aliasing            # TODO: add a note
                 -fpermissive                    # Downgrade some diagnostics about nonconformant code from errors to warnings.
-                -march=core-avx2                # Specify the name of the target architecture
+                $<IF:$<STREQUAL:${CMAKE_SYSTEM_PROCESSOR},aarch64>,-march=armv8.2-a,-march=core-avx2>                # Specify the name of the target architecture
                 -pipe                           # Use pipes rather than intermediate files.
                 -pthread                        # Define additional macros required for using the POSIX threads library.
                 -w                              # Inhibit all warning messages.
