diff --git a/cmake/McrtMessagesCompileOptions.cmake b/cmake/McrtMessagesCompileOptions.cmake
index 9f6bc76..5c3e59b 100644
--- a/cmake/McrtMessagesCompileOptions.cmake
+++ b/cmake/McrtMessagesCompileOptions.cmake
@@ -12,7 +12,7 @@ function(${PROJECT_NAME}_cxx_compile_options target)
                 -fno-omit-frame-pointer         # TODO: add a note
                 -fno-strict-aliasing            # TODO: add a note
                 -fpermissive                    # Downgrade some diagnostics about nonconformant code from errors to warnings.
-                -march=core-avx2                # Specify the name of the target architecture
+                $<IF:$<STREQUAL:${CMAKE_SYSTEM_PROCESSOR},aarch64>,-march=armv8.2-a,-march=core-avx2>                # Specify the name of the target architecture
                 -pipe                           # Use pipes rather than intermediate files.
                 -pthread                        # Define additional macros required for using the POSIX threads library.
                 -w                              # Inhibit all warning messages.
